File: wlmedia_sample/wlmedia/src/main/java/com/ywl5320/wlmedia/util/WlColorUtil.java
package com.ywl5320.wlmedia.util;

/**
 * author : ywl5320
 * e-mail : ywl5320@163.com
 * desc   : wlmedia
 * date   : 2024/8/10
 */
public class WlColorUtil {
    public static boolean isRGBAColor(String color) {
        if (color == null) {
            return false;
        }
        if (!color.startsWith("#")) {
            return false;
        }
        if (color.length() != 9 && color.length() != 7) {
            return false;
        }
        for (int i = 1; i < color.length(); i++) {
            char c = color.charAt(i);
            if (!Character.isDigit(c) && (c < 'a' || c > 'f') && (c < 'A' || c > 'F')) {
                return false;
            }
        }
        return true;
    }
}

File: platform/wlmedia_hm/libwlmedia/src/main/ets/com/ywl5320/wlmedia/util/WlColorUtil.ts
/**
 * author : ywl5320
 * e-mail : ywl5320@163.com
 * desc   : wlmedia
 * date   : 2025/2/4
 */
export class WlColorUtil {
  public static isRGBAColor(color: string | null): boolean {
    if (color === null) {
      return false;
    }
    if (!color.startsWith("#")) {
      return false;
    }
    if (color.length !== 9 && color.length !== 7) {
      return false;
    }
    for (let i = 1; i < color.length; i++) {
      const c = color.charAt(i);
      if (!this.isHexChar(c)) {
        return false;
      }
    }
    return true;
  }

  private static isHexChar(c: string): boolean {
    return /[0-9a-fA-F]/.test(c);
  }
}